tests/test_slim_engine.py

```diff
@@ -1420,17 +1422,118 @@ def test_no_mutation_types_defined(self):
                 dry_run=True,
             )
 
+    def test_multiple_mutation_types_defined(self):
+        extended_events = [
+            stdpopsim.ext.DrawMutation(
+                time=self.T_mut,
+                mutation_id="test",
+                population_id=0,
+                coordinate=100,
+            ),
+        ]
+        contig = get_test_contig()
+        mt = stdpopsim.MutationType(
+            distribution_type="f",
+            dominance_coeff=1.0,
+            distribution_args=[0.0],
+            convert_to_substitution=False,
+        )
+        dfe = stdpopsim.DFE(
+            id="test",
+            mutation_types=[mt, mt],
+            proportions=[0.5, 0.5],
+            description="test test",
+            long_description="test test test",
+        )
+        contig.add_dfe(
+            intervals=np.empty((0, 2), dtype="int"),
+            DFE=dfe,
+        )
+        engine = stdpopsim.get_engine("slim")
+        with pytest.raises(ValueError):
```

Comment:
We should be checking we're actually throwing the right ValueError here, like:
```suggestion
        with pytest.raises(ValueError, match=<some substring of the error message we're trying to hit>):
```